test/unit/TicketBroker.js

```diff
@@ -620,4 +648,245 @@ contract("TicketBroker", accounts => {
+            const sender2 = accounts[2]
+            await broker.fundDeposit({from: sender, value: 1000})
+            await broker.fundDeposit({from: sender2, value: 2000})
+            await broker.unlock({from: sender})
+            await broker.unlock({from: sender2})
+
+            await expectRevertWithReason(broker.unlock({from: sender}), "unlock already initiated")
+            await expectRevertWithReason(broker.unlock({from: sender2}), "unlock already initiated")
+        })
+
+        it("sets withdrawBlock according to constructor config", async () => {
+            await broker.fundDeposit({from: sender, value: 1000})
+            const fromBlock = (await web3.eth.getBlock("latest")).number
+
+            await broker.unlock({from: sender})
+
+            const withdrawBlock = (await broker.senders.call(sender)).withdrawBlock.toString()
+            const expectedWithdrawBlock = fromBlock + unlockPeriod + 1 // +1 to account for the block created executing unlock
+            assert.equal(withdrawBlock, expectedWithdrawBlock.toString())
+        })
+
+        it("sets isUnlockInProgress to true", async () => {
+            await broker.fundDeposit({from: sender, value: 1000})
+
+            await broker.unlock({from: sender})
+
+            const isUnlockInProgress = await broker.isUnlockInProgress.call(sender)
+            assert(isUnlockInProgress)
+        })
+
+        it("emits an Unlock event", async () => {
+            await broker.fundDeposit({from: sender, value: 1000})
+
+            const txResult = await broker.unlock({from: sender})
+
+            truffleAssert.eventEmitted(txResult, "Unlock", ev => {
+                return ev.sender === sender
+            })
+        })
+
+        it("emits an Unlock event indexed by sender", async () => {
+            const fromBlock = (await web3.eth.getBlock("latest")).number
+            await broker.fundDeposit({from: sender, value: 1000})
+
+            await broker.unlock({from: sender})
+
+            const events = await broker.getPastEvents("Unlock", {
+                filter: {
+                    sender
+                },
+                fromBlock,
+                toBlock: "latest"
+            })
+            assert.equal(events.length, 1)
+            const event = events[0]
+            assert.equal(event.returnValues.sender, sender)
+        })
+    })
+
+    describe("cancelUnlock", () => {
+        it("reverts if sender is not in an unlocking state", async () => {
+            await expectRevertWithReason(broker.cancelUnlock(), "no unlock request in progress")
+        })
+
+        it("sets isUnlockInProgress to false", async () => {
+            await broker.fundDeposit({from: sender, value: 1000})
+            await broker.unlock()
+
+            await broker.cancelUnlock()
+
+            const isUnlockInProgress = await broker.isUnlockInProgress.call(sender)
+            assert.equal(isUnlockInProgress, false)
```

Comment:
`assert(!isUnlockInProgress)`?

Reply:
totally.. I realized this mistake some time after and forgot to fix!